### ISSUE TITLE
add file to .gitattributes manually sometimes can not track the file

### ISSUE THREAD

**Issue by canyou2014** [git-lfs/git-lfs#2412]
I want to transfer some files to LFS, and do like this: 

```
$ git lfs track bin/cat.obj
Tracking bin/cat.obj
$ git status       #mark here
On branch binary
Your branch is up-to-date with 'origin/binary'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   bin/cat.obj

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        .gitattributes


```
And then I add `*.exe filter=lfs diff=lfs merge=lfs -text` to .gitattributes
```
$ cat .gitattributes 
bin/cat.obj filter=lfs diff=lfs merge=lfs -text
*.exe filter=lfs diff=lfs merge=lfs -text
```
But now , when I run `git status`, it shows:
```
$ git status
On branch binary
Your branch is up-to-date with 'origin/binary'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   bin/cat.obj
        modified:  .gitattributes
```
Some .exe file are not tracked successfully.
But when I do not run the first `git status`(the note "mark here"), all the .exe files can be tracked successfully. Like this:
```
$ git status
On branch binary
Your branch is up-to-date with 'origin/binary'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        modified:   bin/cat.obj
        modified:   bin/bzdiff.exe
        modified:   bin/sed.exe
```
How it works?

**Comment by ttaylorr** [comment]
Hi @canyou2014, thanks for opening this issue and sorry that you're having trouble! Git LFS works by running files through "filters" to convert them between their large contents and the pointer files stored in your history. Git invokes LFS to run these filters based on two things: the contents of the .gitattributes, and the last known `stat(1)` of files that match those filters.

If a file is "stat-clean" (i.e., hasn't changed since the last time it was recorded in history), Git skips it when checking which files to run through the filters. I think what you're experiencing here is some of that.

Luckily, `git-lfs-track(1)` automatically marks each file tracked as "stat-dirty", so Git will reclaim and run it through the filters correctly when you run that command.

In other words, it is discouraged to edit your .gitattributes file directly, unless you're doing something that the `track` command doesn't support.

If you run `git lfs track "*.exe"`, are the files tracked as expected?

**Comment by canyou2014** [comment]
@ttaylorr `git lfs track` track the files successfully for me. I have already understood how it work that , thanks for your explanation.
